DateTime.js

```diff
@@ -68,7 +69,11 @@ var Datetime = createClass({
 		if( state.open == undefined )
 			state.open = !this.props.input;
 
+		if (this.props.onOpen) {
```

Comment:
feels a bit weird that  this happens in `getInitialState`

sounds more appropriate for this to be in one of the component's lifecycle methods

Reply:
no no, this is the old react, getInitialState is what they had instead of the constructor